### PR TITLE
[MM-11282] Migrate audit_table to use redux and be pure

### PR DESCRIPTION
**Pull request by R-Wang97**
#### Summary
Changes necessary to make `audit_table.jsx` use redux and be pure. New tests have been added as well.

#### Ticket Link
https://github.com/mattermost/mattermost-server/issues/9090
https://mattermost.atlassian.net/browse/MM-11282

#### Checklist
- [x] Ran `make check-style` to check for style errors (required for all pull requests)
- [x] Ran `make test` to ensure unit and component tests passed
- [x] Added or updated unit tests (required for all new features)
- [x] Has redux changes (https://github.com/mattermost/mattermost-redux/pull/628)

### CODE REVIEW

File: components/audit_table/audit_table.jsx [hunk 1]
```diff
@@ -218,16 +216,13 @@ function AuditTable(props) {
     const {formatMessage} = props.intl;
     for (var i = 0; i < props.audits.length; i++) {
         const audit = props.audits[i];
```
Comment (saturninoabril) [inline components/audit_table/audit_table.jsx]: Looks like `props.getByName` and `props.getUser` are being called on render function.

I'm imagining that each `audit` should be handled by a new component (maybe `AuditItem`) where user and channel object will be handled by its container.

File: components/audit_table/audit_table.jsx [hunk 2]
```diff
@@ -330,11 +325,14 @@ AuditTable.propTypes = {
...
     showSession: PropTypes.bool,
+    currentUser: PropTypes.object.isRequired,
+    getUser: PropTypes.func.isRequired,
+    getByName: PropTypes.func.isRequired,
 };
 
 export default injectIntl(AuditTable);
 
-export function formatAuditInfo(audit, formatMessage) {
+export function formatAuditInfo(audit, formatMessage, currentUser, getByName, getUser) {
```
Comment (saturninoabril) [inline components/audit_table/audit_table.jsx]: I think we need to start breaking this component like per if condition below,:
- new `AuditChannelInfo` for `actionURL.indexOf('/channels') === 0`
- new `AuditOAuthInfo` `actionURL.indexOf('/oauth') === 0`
- new `AuditUserInfo` for `actionURL.indexOf('/users') === 0`
- new `AuditTimestampInfo` for `audit.create_at`

**Comment by R-Wang97** [comment]
Hey @saturninoabril, could we perhaps make a separate ticket for breaking `audit_table` into smaller components? I'm back in school now and I'm not sure if I'll have the time to do the refactoring that you suggested.

**Comment by saturninoabril** [comment]
Thanks @R-Wang97, ticket created https://mattermost.atlassian.net/browse/MM-12200

**Comment by jasonblais** [comment]
Thanks @R-Wang97! @jwilander looks like this one is ready for your review

**Pull request merged**